test/unit/models/users.test.js

```diff
@@ -221,6 +221,37 @@ describe("users", function () {
     });
   });
 
+  describe("archive user if not in discord", function () {
+    beforeEach(async function () {
+      const addUsersPromises = [];
+      userDataArray.forEach((user) => {
+        const userData = {
+          ...user,
+          roles: {
+            ...user.roles,
+            in_discord: false,
+            archived: false,
+          },
+        };
+        addUsersPromises.push(userModel.add(userData));
+      });
+
+      await Promise.all(addUsersPromises);
+    });
+
```

Comment:
also write a test to check if an error occurs, i.e. if the code goes to catch block